File: froom-backend/src/main/kotlin/com/froom/froombackend/items/model/domain/Category.kt
package com.froom.froombackend.items.model.domain

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

enum class Category(name: String, val bodyPart: BodyPart, val index: Int) {
    TSHIRT("TShirt", BodyPart.TOP, 0),
    TROUSERS("Trousers", BodyPart.BOTTOM, 1),
    PULLOVER("Pullover", BodyPart.TOP, 2),
    DRESS("Dress", BodyPart.TOP, 3),
    COAT("Coat", BodyPart.TOP, 4),
    SANDALS("Sandals", BodyPart.SHOES, 5),
    SHIRT("Shirt", BodyPart.TOP, 6),
    SNEAKERS("Sneakers", BodyPart.SHOES, 7),
    BAG("Bag", BodyPart.ACCESSORY, 8),
    ANKLE_BOOTS("Ankle Boots", BodyPart.SHOES, 9),
    UNKNOWN("Unknown", BodyPart.UNKNOWN, -1);

    @JsonValue
    fun toJson(): String {
        return name
    }

    companion object {
        @JsonCreator
        @JvmStatic
        fun fromJson(name: String): Category {
            return entries.find { it.name.equals(name, ignoreCase = true) } ?: throw IllegalArgumentException("Unknown category: $name")
        }
    }

}

File: froom-frontend/src/model/enums/Category.ts
import {BodyPart} from './BodyPart.ts';

enum Category {
    TSHIRT = 'TShirt',
    TROUSERS = 'Trousers',
    PULLOVER = 'Pullover',
    DRESS = 'Dress',
    COAT = 'Coat',
    SANDALS = 'Sandals',
    SHIRT = 'Shirt',
    SNEAKERS = 'Sneakers',
    BAG = 'Bag',
    ANKLE_BOOTS = 'Ankle Boots',
    UNKNOWN = 'Unknown'
}

const categoryToBodyPart = new Map<Category, BodyPart>([
    [Category.TSHIRT, BodyPart.TOP],
    [Category.TROUSERS, BodyPart.BOTTOM],
    [Category.PULLOVER, BodyPart.TOP],
    [Category.DRESS, BodyPart.TOP],
    [Category.COAT, BodyPart.TOP],
    [Category.SANDALS, BodyPart.SHOES],
    [Category.SHIRT, BodyPart.TOP],
    [Category.SNEAKERS, BodyPart.SHOES],
    [Category.BAG, BodyPart.ACCESSORY],
    [Category.ANKLE_BOOTS, BodyPart.SHOES],
    [Category.UNKNOWN, BodyPart.UNKNOWN]
]);


export {Category, categoryToBodyPart};
